core/trie2/databasetest.go

```diff
@@ -0,0 +1,189 @@
+package trie2
+
+import (
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/core/trie2/triedb/database"
+	"github.com/NethermindEth/juno/core/trie2/trienode"
+	"github.com/NethermindEth/juno/core/trie2/trieutils"
+	"github.com/NethermindEth/juno/db"
+)
+
+type dbScheme uint8
+
+const (
+	PathScheme dbScheme = iota + 1
+	HashScheme
+)
+
+type testNodeReader struct {
```

Comment:
`nit`: I feel like there has to be a better name than `databasetest`, but I can't think of any.